src/config/config.ts

```diff
@@ -1,8 +1,43 @@
-import {Injectable} from '@angular/core';
+import {Injectable, EventEmitter, ChangeDetectorRef} from '@angular/core';
 
 @Injectable()
 export class NglConfig {
 
   svgPath = 'assets/icons';
 
+  private _emitter = new EventEmitter();
+
+  refresh() {
+    this._emitter.emit();
+  }
+
+  _attach(cd: ChangeDetectorRef) {
+    return this._emitter.subscribe(() => cd.markForCheck());
+  }
 }
+
+// Intrenal decorator
+export function NglConfigurable() {
+  return function (constructor: Function) {
+    let { ngOnInit, ngOnDestroy } = constructor.prototype;
+
+    constructor.prototype.ngOnInit = function() {
+      this.configSubscription = this.config._attach(this.cd);
```

Comment:
Here we cannot be sure about the `ChangeDetectorRef1 instance name. As such, I would suggest:
1. Require the name of the instance in an object received by the `NglConfigurable` decorator
2. Throw an error if the property is not there or, if the type is not the expected one